### ISSUE TITLE
feature: C# 9 records serialization

### ISSUE THREAD

**Issue by roubachof** [reactiveui/Akavache#694]
C#9 records are finally the immutable type that we needed in C#.
Since its syntax is game changing, akavache should supports serializing/deserializing such objects.
Could be solved by #666.

**Comment by glennawatson** [comment]
Yeah #666 will happen next major release. Will be in the next few weeks hopefully.

**Comment by rezamohamed** [comment]
How would one change the serializer? to say MessgePack?

**Comment by glennawatson** [comment]
In the upcoming version there is a IAkavacheSerializer you register with Splat.

**Comment by Syed-RI** [comment]
Hmm any movements here? Knock! Knock!

**Comment by ChrisPulman** [comment]
Hi we have a pending release of V11.0.x this is a ground up rewrite, there will be a migration process but we are hopeful that the new version will have a better experience, I have added many tests based on some of the issues I have seen raised to hopefully resolve the issues that previous versions have experienced. #1052 has the latest documentation for the pending release.

**Comment by ChrisPulman** [comment]
V11.0.1 has now been released, please if this now solves your Serialisation issue, please update us and if relevant, close this issue. 
Many thanks.

**Comment by Syed-RI** [comment]
I think this can be closed now with a link to the docs for v11.xx versions! Thanks for all the work team ❤️

**Comment by ChrisPulman** [comment]
https://github.com/reactiveui/Akavache#serializers

<blockquote><img src="https://repository-images.githubusercontent.com/2896461/66714280-b19a-11ea-9182-63486a2fb10e" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/reactiveui/Akavache">GitHub - reactiveui/Akavache: An asynchronous, persistent key-value store created for writing desktop and mobile applications, based on SQLite3. Akavache is great for both storing important data as well as cached local data that expires.</a></strong></div><div>An asynchronous, persistent key-value store created for writing desktop and mobile applications, based on SQLite3. Akavache is great for both storing important data as well as cached local data tha...</div></blockquote>
